python/cudf/cudf/core/dataframe.py

```diff
@@ -272,7 +270,11 @@ def _init_from_dict_like(self, data, index=None, columns=None):
         for (i, col_name) in enumerate(data):
             self.insert(i, col_name, data[col_name])
 
-        self.columns = columns
+        if columns is not None:
+            self.columns = columns
+
+        if len(data) and all(isinstance(x, tuple) for x in data):
```

Comment:
Is it a safe assumption that if all the names are tuples that it's a multiindex?

Reply:
When constructing a DataFrame from a dictionary, that's the assumption Pandas makes..

Reply:
Could we fuse this loop with the enumerated loop on data above so we prevent having to iterate over the columns twice?

Reply:
Done!